ponzu/llama/curatedTVL.py

```diff
@@ -0,0 +1,130 @@
+from llama.llama import Llama
+import pandas as pd
+
+class CuratedTVL(Llama):
+    def __init__(self):
+        super().__init__()  # call Llama's __init__ method
+        self.tvl = None
+        self.raw_tvl = None
+        self.curated_tvl = None
+
+    def fetch_tvl(self, chains):
+        if self.raw_tvl is None:
+            self.raw_tvl = self.getChainTVL(chains)
```

Comment:
i pulled out some of the state (dfs) saving stuff in the base llama bc 1) I wasn't using it downstream and 2) it basically doubled the memory requirements which proved to be a bottleneck on hex

Reply:
so if you dont really have a good use case to save it to the class, i'd save to a local var and return

Reply:
fed chat this comment and it's saying that my file is managing memory effectively. lmk if u don't think that's the case tho and can walk me thru what it looks like to save as a local var vs have in class

Reply:
we can test it out and see if it plays nicely with hex -- i just know i had the same pattern (saving the df to a class var) and was able to cut the memory load on hex by half when i removed it (bc you are storing it in the class var and the variable you assign in the notebook like df = peterFunction()